c7n/policy.py

```diff
@@ -91,10 +91,11 @@ def from_data(cls, data: dict, options, session_factory=None):
     def __add__(self, other):
         return self.__class__(self.policies + other.policies, self.options)
 
-    def filter(self, policy_patterns=[], resource_types=[]):
+    def filter(self, policy_patterns=[], resource_types=[], providers=[], modes=[]):
```

Comment:
providers unused, also can we switch the method defaults here to tuples, mutable as default args is typically an err.

Reply:
Yea I can drop providers, modes was sufficient since provider is implied